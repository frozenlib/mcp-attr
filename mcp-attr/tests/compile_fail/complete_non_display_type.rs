use mcp_attr::{server::{mcp_server, McpServer, RequestContext}, Result};

struct TestServer;

#[derive(Debug)]
struct NonDisplayType {
    value: i32,
}

async fn complete_non_display(_value: &str, _cx: &RequestContext) -> Result<Vec<NonDisplayType>> {
    Ok(vec![NonDisplayType { value: 42 }])
}

#[mcp_server]
impl McpServer for TestServer {
    #[prompt]
    async fn test_prompt(&self, #[complete(complete_non_display)] msg: String) -> Result<String> {
        Ok(msg)
    }
}

fn main() {}