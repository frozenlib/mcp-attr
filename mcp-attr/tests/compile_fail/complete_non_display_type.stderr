error[E0277]: `NonDisplayType` doesn't implement `std::fmt::Display`
  --> tests/compile_fail/complete_non_display_type.rs:11:10
   |
11 | async fn complete_non_display(_value: &str, _cx: &RequestContext) -> Result<Vec<NonDisplayType>> {
   |          ^^^^^^^^^^^^^^^^^^^^ `NonDisplayType` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `NonDisplayType`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: required for `mcp_attr::schema::CompleteResult` to implement `From<Vec<NonDisplayType>>`
   = note: required for `Vec<NonDisplayType>` to implement `Into<mcp_attr::schema::CompleteResult>`
